python/lsst/obs/auxTel/auxTelMapper.py

```diff
@@ -0,0 +1,338 @@
+
+        @param format  The desired set of keys
+        @param dataId  A possible-incomplete dataId
+        """
+        nChannel = 16                   # number of possible channels, 1..nChannel
+
+        if "channel" in dataId:         # they specified a channel
+            dataId = dataId.copy()
+            channels = [dataId.pop('channel')]
+        else:
+            channels = range(1, nChannel + 1)  # we want all possible channels
+
+        if "channel" in format:           # they asked for a channel, but we mustn't query for it
+            format = list(format)
+            channelIndex = format.index('channel')  # where channel values should go
+            format.pop(channelIndex)
+        else:
+            channelIndex = None
+
+        dids = []                       # returned list of dataIds
+        for value in self.query_raw(format, dataId):
+            if channelIndex is None:
+                dids.append(value)
+            else:
+                for c in channels:
+                    did = list(value)
+                    did.insert(channelIndex, c)
+                    dids.append(tuple(did))
+
+        return dids
+    #
+    # The composite type "raw" doesn't provide e.g. query_raw, so we defined type _raw in the .paf file
+    # with the same template, and forward requests as necessary
+    #
+
+    def query_raw(self, *args, **kwargs):
+        """Magic method that is called automatically if it exists.
+
+        This code redirects the call to the right place, necessary because of leading underscore on _raw.
+        """
+        return self.query__raw(*args, **kwargs)
+
+    def map_raw_md(self, *args, **kwargs):
+        """Magic method that is called automatically if it exists.
+
+        This code redirects the call to the right place, necessary because of leading underscore on _raw.
+        """
+        return self.map__raw_md(*args, **kwargs)
+
+    def map_raw_filename(self, *args, **kwargs):
+        """Magic method that is called automatically if it exists.
+
+        This code redirects the call to the right place, necessary because of leading underscore on _raw.
+        """
+        return self.map__raw_filename(*args, **kwargs)
+
+    def bypass_raw_filename(self, *args, **kwargs):
+        """Magic method that is called automatically if it exists.
+
+        This code redirects the call to the right place, necessary because of leading underscore on _raw.
+        """
+        return self.bypass__raw_filename(*args, **kwargs)
+
+    def map_raw_visitInfo(self, *args, **kwargs):
+        """Magic method that is called automatically if it exists.
+
+        This code redirects the call to the right place, necessary because of leading underscore on _raw.
+        """
+        return self.map__raw_visitInfo(*args, **kwargs)
+
+    def bypass_raw_visitInfo(self, datasetType, pythonType, location, dataId):
+        if False:
```

Comment:
It doesn't feel right to have this here.  I think this block should be made into a Jira issue this `if False` block removed.

Reply:
I have just removed it completely for now, I think that's OK. Shout if you want something different to happen here.